src/Operations.jl

```diff
@@ -932,11 +932,12 @@ function rm(ctx::Context, pkgs::Vector{PackageSpec})
     write_env(ctx.env)
 end
 
-update_package_add(pkg::PackageSpec, ::Nothing, is_dep::Bool) = pkg
-function update_package_add(pkg::PackageSpec, entry::PackageEntry, is_dep::Bool)
+update_package_add(ctx::Context, pkg::PackageSpec, ::Nothing, is_dep::Bool) = pkg
+function update_package_add(ctx::Context, pkg::PackageSpec, entry::PackageEntry, is_dep::Bool)
     if entry.pinned
-        pkg.version == VersionSpec() ||
-            @warn "`$(pkg.name)` is pinned at `v$(entry.version)`. Maintaining pinned version."
+        if pkg.version == VersionSpec()
+            println(ctx.io, "`$(pkg.name)` is pinned at `v$(entry.version)`: maintaining pinned version")
```

Comment:
We should probably make our own `pkgwarning` or something because I like that you get the color and `WARNING:` with `@warn`. Now it will just be standard text.

Reply:
Doesnt have to be done here though.